### ISSUE TITLE
FxMixer: sends to moved channels are not saved/loaded properly

### ISSUE THREAD

**Issue by Fastigium** [LMMS/lmms#2627]
Steps to reproduce:

1. Start with an empty project
2. Add 3 FX channels
3. Make FX 1 send to FX 3
4. Move FX 3 to the left, either using the context menu or by selecting it and pressing Alt + Left
5. Select FX 1 and confirm that it sends to the channel directly to the right of it now
6. Save the project and load it
7. Select FX 1 and notice that it sends to the channel *next to* the channel to the right of it now

This bug made half of my kick end up on the hi-hat channel in the track I'm making in `master` :sweat_smile:. It is not present in `stable-1.1`.

**Comment by zonkmachine** [comment]
```
$ git bisect bad
e91991216df9b2fc260c5c80bb563140459d3b0b is the first bad commit
commit e91991216df9b2fc260c5c80bb563140459d3b0b
Author: Amadeus Folego <amadeusfolego@gmail.com>
Date:   Fri Jan 23 05:12:48 2015 -0200

    Fix segfault when moving channels
    
    This commit addresses a lot of issues, namely:
    
    1. When the "Move left/right" action was selected on a mixer channel
    menu sometimes it would segfault due to the fxline object being
    deleted before it returned to a method inside itself
    
    2. The Fader was declaring a new variable for the model whereas it
    should used the inherited model from FloatModelView < AutomatableModelView
    
    3. Some methods were doing more things than they should be and
    performing unnecessary actions. A little cleanup/refactor was made
    
    Please notice that a bug of the same class as the one referred at point 1
    still exists: clicking on "Remove channel". This commit does not
    addresses this issue.
    
    Fixes #1584
``` 

:tractor:

**Comment by tresf** [comment]
Thanks for the investigation.

Although no longer active on our tracker, @badosu is still around and usually hangs out on IRC #lmms.  I'll ping him. :+1:

**Comment by badosu** [comment]
@zonkmachine @Fastigium I am not properly setup to develop lmms these days, but let me know if I can help in any way

**Comment by Fastigium** [comment]
Pull request made: #2641. This fixes the test case for me. Testing welcome!

@zonkmachine Thanks for the bisect work :tractor:, that pointed me in the right direction.
@badosu If you could look the pull request over and give any comments/suggestions that come to mind, I'd appreciate it :+1:

**Comment by zonkmachine** [comment]
#2641 Tested, works! :+1:

**Comment by tresf** [comment]
Closed via 9ff8091.
